app/mm-redux/selectors/entities/channel_categories.ts

```diff
@@ -327,49 +327,16 @@ export function makeFilterAndSortChannelsForCategory() {
 
 export function makeGetChannelsByCategory() {
     const getCategoriesForTeam = makeGetCategoriesForTeam();
-    const getUnsortedUnfilteredChannels = makeGetUnsortedUnfilteredChannels();
-
-    // Memoize filterAndSortChannels by category. As long as the categories don't change, we can keep using the same
-    // selector for each category.
-    let filterAndSortChannels: RelationOneToOne<ChannelCategory, ReturnType<typeof makeFilterAndSortChannelsForCategory>>;
-
-    let lastCategoryIds: ReturnType<typeof getCategoryIdsForTeam> = [];
-    let lastChannelsByCategory: RelationOneToOne<ChannelCategory, Channel[]> = {};
 
     return (state: GlobalState, teamId: string) => {
-        const categoryIds = getCategoryIdsForTeam(state, teamId);
-
-        // Create an instance of filterAndSortChannels for each category. As long as we don't add or remove new categories,
-        // we can reuse these selectors to memoize the results of each category. This will also create new selectors when
-        // categories are reordered, but that should be rare enough that it won't meaningfully affect performance.
-        if (categoryIds !== lastCategoryIds) {
-            lastCategoryIds = categoryIds;
-            lastChannelsByCategory = {};
-
-            filterAndSortChannels = {};
-
-            if (categoryIds) {
-                for (const categoryId of categoryIds) {
-                    filterAndSortChannels[categoryId] = makeFilterAndSortChannelsForCategory();
-                }
-            }
-        }
-
         const categories = getCategoriesForTeam(state, teamId);
-        const channels = getUnsortedUnfilteredChannels(state, teamId);
-
-        const channelsByCategory: RelationOneToOne<ChannelCategory, Channel[]> = {};
+        const unreadChannels = getUnreadChannelIds(state);
 
-        for (const category of categories) {
-            channelsByCategory[category.id] = filterAndSortChannels[category.id](state, channels, category);
-        }
-
-        // Do a shallow equality check of channelsByCategory to avoid returning a new object containing the same data
-        if (shallowEquals(channelsByCategory, lastChannelsByCategory)) {
-            return lastChannelsByCategory;
-        }
+        // Remove unreads from categories
+        categories.forEach((category) => {
+            category.channel_ids = category.channel_ids.filter((channel) => !unreadChannels.includes(channel));
+        });
```

Comment:
Realised that the categories endpoint returns things in order, with only the unreads being out of place as per design. For filtering - mobile app uses the existing filterList.

Reply:
What happens when I switch the order on the webapp and then open the mobile app and what happens if I change the order of things in the webapp while the mobile app is in used or in the background.